frameworks/Java/quarkus/pgclient/src/main/java/io/quarkus/benchmark/repository/pgclient/PgClients.java

```diff
@@ -0,0 +1,36 @@
+package io.quarkus.benchmark.repository.pgclient;
+
+import io.vertx.mutiny.pgclient.PgPool;
+import io.vertx.mutiny.sqlclient.SqlClient;
+
+class PgClients {
+    private ThreadLocal<SqlClient> sqlClient = new ThreadLocal<>();
+    private ThreadLocal<PgPool> pool = new ThreadLocal<>();
+    private PgClientFactory pgClientFactory;
+
+	// for ArC
+	public PgClients() {
+	}
+
+	public PgClients(PgClientFactory pgClientFactory) {
+	    this.pgClientFactory = pgClientFactory;
+    }
+
+    SqlClient getClient() {
+        SqlClient ret = sqlClient.get();
+        if(ret == null) {
+            ret = pgClientFactory.sqlClient(1);
+            sqlClient.set(ret);
+        }
+		return ret;
+	}
+
+	synchronized PgPool getPool() {
+        PgPool ret = pool.get();
+        if(ret == null) {
+            ret = pgClientFactory.sqlClient(4);
```

Comment:
would be nice for the 4 to be a constant for readability

Reply:
Done.